### PR TITLE
Add release blog Knative 1.14

### PR DESCRIPTION
**Pull request by dsimansk**
## Proposed Changes 

- Add release blog Knative 1.14

/cc @knative/knative-release-leads

### CODE REVIEW

File: blog/docs/releases/announcing-knative-v1-14-release.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+- EventType Autocreate is now a non blocking operation ([#7709](https://github.com/knative/eventing/pull/7709), @Cali0707)
+- EventTypes are now autocreated on Triggers and Subscriptions when there is a reply event sent to them ([#7733](https://github.com/knative/eventing/pull/7733), @Cali0707)
+- Feature: Added the ability for users to configure `nodeSelector` when deploying `apiserversource` , Users can now configure nodeSelector via config-features by adding key-value pair in the format apiserversources.nodeselector.<key>: <value> ([#7584](https://github.com/knative/eventing/pull/7584), @sadath-12)
+- Merge keys, which were associated with an older version of YAML (1.1), are no longer officially supported in YAML according to its specifications. YAML version 1.2, the current standard, does not include support for merge keys, rendering them obsolete. ([#7662](https://github.com/knative/eventing/pull/7662), @converge)
+- StatefulSet scheduling now makes fewer API server requests, reducing APIServer load. ([#7651](https://github.com/knative/eventing/pull/7651), @Cali0707)
+- Using 2.15.2 of Go-sdk for CloudEvents ([#7747](https://github.com/knative/eventing/pull/7747), @matzew)
```
Comment (creydr) [inline blog/docs/releases/announcing-knative-v1-14-release.md:50]:
```suggestion
- EventType Autocreate is now a non blocking operation ([#7709](https://github.com/knative/eventing/pull/7709), @Cali0707)
- EventTypes are now autocreated on Triggers and Subscriptions when there is a reply event sent to them ([#7733](https://github.com/knative/eventing/pull/7733), @Cali0707)
- Added the ability to configure a `nodeSelector` when deploying an ApiServerSource ([#7584](https://github.com/knative/eventing/pull/7584), @sadath-12)
- Replace YAML merge tags, to comply with YAML 1.2 ([#7662](https://github.com/knative/eventing/pull/7662), @converge)
- StatefulSet scheduling now makes fewer API server requests, reducing APIServer load. ([#7651](https://github.com/knative/eventing/pull/7651), @Cali0707)
```

Removed the topics which are not relevant for the users

**Review by Cali0707 — APPROVED** [review]
/lgtm from my side

/hold for @izabelacg

**Pull request merged**